arangod/RocksDBEngine/RocksDBEdgeIndex.cpp

```diff
@@ -355,7 +362,28 @@ class RocksDBEdgeIndexLookupIterator final : public IndexIterator {
             incrCacheHits();
             needRocksLookup = false;
             // We got sth. in the cache
-            VPackSlice cachedData(finding.value()->value());
+            uint8_t const* data = finding.value()->value();
+            if (data[0] == 0xffU) {
+              TRI_ASSERT(finding.value()->size() >= 5);
+              // Custom type. this means the data is lz4-compressed
+              uint32_t uncompressedSize;
+              memcpy(&uncompressedSize, data + 1, sizeof(uncompressedSize));
+              uncompressedSize = basics::bigToHost<uint32_t>(uncompressedSize);
+              if (_lz4CompressBuffer.size() < uncompressedSize) {
+                _lz4CompressBuffer.resize(uncompressedSize);
```

Comment:
Is it somehow clear that this code is only used from a single thread? The buffer is local to the object, so this is a question about multithreaded-use of this class.

Reply:
Yes, an IndexIterator is supposed to be used by only a single thread.
This is true for all other IndexIterator implementations as well. They are all not thread-safe (because there is no need to make them thread-safe right now).